python/ray/data/datasource/file_metadata_shuffler.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Dict, List, Tuple
+
+
+class FileMetadataShuffler:
+    """Abstract class for file metadata shuffler.
+
+    Shufflers live on the driver side of the Dataset only.
+    """
+
+    def __init__(self, reader_args: Dict[str, Any]):
+        self._reader_args = reader_args
+
+    def shuffle_files(
```

Comment:
nit; similar to class name update, should we update the method name + docstrings to something like `shuffle_file_metadatas` or `shuffle_metadatas`? Is it possibly confusing with shuffling files instead of their metadata?